language/evm/move-to-yul/src/solidity_ty.rs

```diff
@@ -0,0 +1,627 @@
+    Uint(usize),
+    Int(usize),
+    Fixed(usize, usize),
+    Ufixed(usize, usize),
+    Address(bool),
+}
+
+/// Represents a Solidity type
+/// TODO: struct
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone)]
+pub(crate) enum SolidityType {
+    Primitive(SolidityPrimitiveType),
+    Tuple(Vec<SolidityType>),
+    DynamicArray(Box<SolidityType>),
+    StaticArray(Box<SolidityType>, usize),
+    SolidityString,
+    Bytes,
+    BytesStatic(usize),
+}
+
+// ================================================================================================
+// Pretty print for SoliditySignature
+
+impl fmt::Display for SoliditySignature {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "sig:{} ", self.sig_name)?;
+        for (ty, data_location) in &self.para_types {
+            write!(f, " ty:{}, location:{} ", ty, data_location)?;
+        }
+        for (ty, data_location) in &self.ret_types {
+            write!(f, " ty:{}, location:{} ", ty, data_location)?;
+        }
+        f.write_str("")
+    }
+}
+
+impl SoliditySignature {
+    /// Generate parameter list for computing the function selector
+    fn compute_param_types(&self, param_types: &[&SolidityType]) -> String {
+        let display_type_slice = |tys: &[&SolidityType]| -> String {
+            tys.iter()
+                .map(|t| format!("{}", t))
+                .collect::<Vec<_>>()
+                .join(",")
+        };
+        display_type_slice(param_types)
+    }
+
+    pub fn selector_signature(&self) -> String {
+        format!(
+            "{}({})",
+            self.sig_name,
+            self.compute_param_types(&self.para_types.iter().map(|(ty, _)| ty).collect_vec())
+        )
+    }
+}
+
+// ================================================================================================
+// Pretty print for SignatureDataLocation
+
+impl fmt::Display for SignatureDataLocation {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        use SignatureDataLocation::*;
+        match self {
+            // CallData => f.write_str("calldata"),
+            Memory => f.write_str("memory"),
+        }
+    }
+}
+
+// ================================================================================================
+// Pretty print for SolidityPrimitiveType
+
+impl fmt::Display for SolidityPrimitiveType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        use SolidityPrimitiveType::*;
+        match self {
+            Bool => f.write_str("bool"),
+            Uint(n) => write!(f, "uint{}", n),
+            Int(n) => write!(f, "int{}", n),
+            Fixed(m, n) => write!(f, "fixed{}x{}", m, n),
+            Ufixed(m, n) => write!(f, "ufixed{}x{}", m, n),
+            Address(_) => f.write_str("address"),
+        }
+    }
+}
+
+// ================================================================================================
+// Pretty print for SolidityType
+
+impl fmt::Display for SolidityType {
```

Comment:
nit: still think this display should produce a string which is correct solidity syntax and can be passed into your parser. Nice for testing but also perhaps some point in error messages or similar. If that is a debug print, you would impl the Debug print trait.